### PR TITLE
ci: automatic create release pull-request; lint commit message

### PR DESCRIPTION
**Pull request by whichwit**
**TL;DR:**
- The only significant formatting convention to pay attention to are PR Title and PR Description, these are converted to commit subject and commit body/footer when the PR is merged to the main branch;
  - `feat: <short-message>` advances minor version
  - every other type advances patch version
  - major version is advanced manually
- Use `Squash and Merge` when merging any PR;
- GitHub Issue formatting is insignificant, as these can be created by the community;
- Messages from commits toward a branch associated with a PR are insignificant; 

**Adapt:**
- [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [Semantic Versioning](https://semver.org/spec/v2.0.0.html)

### CODE REVIEW

**Comment by whichwit** [comment]
This PR is ready for review. Once merged, I will monitor the publishing pipeline to make sure it remains functional.

**Comment by johnarleyburns** [comment]
Looks good, what is missing is README update about what the new PR naming
convention is, and what this new configuration does in auto-creating PRs
and versions.  We just need to thoroughly document it as part of the PR.

**Comment by whichwit** [comment]
Updated README, the contribute section.

**Review by johnarleyburns — APPROVED** [review]
